src/core/utils/CorePlugin.ts

```diff
@@ -20,7 +30,96 @@ export class CorePlugin extends JWPlugin {
     // Public
     //--------------------------------------------------------------------------
 
-    removeSide(intent: Action): void {
+    removeSide(intent: Intent): void {
         console.log('REMOVE SIDE:' + intent);
     }
+    /**
+     * Navigate to a given Range (in the payload of the Intent).
+     *
+     * @param intent
+     */
+    navigate(intent: Intent): void {
+        this._setRange(intent.payload['value']);
+    }
+    /**
+     * Render the `vDocument`.
+     */
+    render(): void {
+        this.vDocument.renderer.render(this.vDocument.editable);
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Return a position in the `VDocument` as a tuple `[VNode, 'before' | 'after']`
+     * (ie. the position is before or after a given `VNode`). The position is
+     * always given on the leaf.
+     *
+     * @param container
+     * @param offset
+     */
+    _getPosition(container: DOMElement | Node, offset: number): VPosition {
+        const containers = this.vDocument.domMap.get(container);
+        const isTextNode = container.nodeType === 3;
+        if (isTextNode) {
+            return this._positionToLeaf({
+                ref: containers[offset] || containers[containers.length - 1],
+                side: offset < containers.length ? 'before' : 'after',
+            });
+        }
+        return this._positionToLeaf({ ref: containers[offset], side: 'before' });
+    }
+    /**
+     * Return a tuple with the range nodes in ltr order (rtl if `reverse` is
+     * true) and updates their `order` property accordingly.
+     *
+     * @param reverse
+     */
+    _orderedRangeNodes(reverse = false): [VNode, VNode] {
+        const rangeNodes = this.vDocument.rangeNodes;
+        const res: [VNode, VNode] = [rangeNodes.start, rangeNodes.end];
+        if (reverse) {
+            res.reverse();
+        }
+        res.forEach((rangeNode: VNode, i: 0 | 1) => {
+            rangeNode.order = i;
```

Comment:
I once attented a conference where the speaker was arguing against using single letters (among others) for variable names as it did not give any indication of what the variable was supposed to represent. One of the guy in the room asked him about the `i` variable in loops. The speaker responded something along the line of "Are you so lazy that you can't call the variable  `index` or, even better, `whateverIndex` depending on your context, since that *is* what that variable is ?" and it rang very true to me. Basically, the only place where I still accept single-letter variables is in very short and obvious constructions like: `keys.forEach(k => k.trim())`. So TL;DR: rename `i` into either `index` or `nodeIndex`... ;-)